homeassistant/components/shelly/switch.py

```diff
@@ -0,0 +1,165 @@
+from homeassistant.core import callback
+
+from .const import (
+    COMMAND_SUFFIX,
+    CONF_MODEL,
+    CONF_TOPIC,
+    DOMAIN,
+    MODEL_SWITCHES,
+    MODELS,
+)
+
+ON = "on"
+OFF = "off"
+TOGGLE = "toggle"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Shelly switch(es) from config entry."""
+
+    device_id = config_entry.data[CONF_DEVICE_ID]
+    topic = config_entry.data[CONF_TOPIC]
+    model = config_entry.data[CONF_MODEL]
+    switches_count = MODELS[model][MODEL_SWITCHES]
+    switches = []
+    for i in range(0, switches_count):
+        switch_topic = topic + f"relay/{i}"
+        if i == 0:
+            unique_id = device_id
+            via_device = None
+        else:
+            unique_id = f"{device_id}_{i+1}"
+            via_device = device_id
+        switches.append(
+            ShellySwitch(
+                hass.components.mqtt, unique_id, switch_topic, model, via_device
+            )
+        )
+
+    async_add_entities(switches)
+
+
+class ShellySwitch(SwitchEntity):
+    """Representation of a Shelly switch."""
+
+    def __init__(self, mqtt, unique_id, topic, model, via_device):
+        """Initialize the MQTT switch."""
+        self._mqtt = mqtt
+        self._unique_id = unique_id
+        self._via_device = via_device
+        self._topic = topic
+        self._model = model
+        self._power = None
+        self._energy = None
+        self._unsubscribe = None
+        self._state = None
+
+    async def async_added_to_hass(self):
+        """Subscribe to MQTT events."""
+        await super().async_added_to_hass()
+        await self._subscribe_topics()
+
+    async def _subscribe_topics(self):
+        """(Re)Subscribe to topics."""
+
+        @callback
+        def state_message_received(msg):
+            """Handle new MQTT state messages."""
+            if msg.topic == self._topic:
+                payload = msg.payload
+                if payload == ON:
+                    self._state = True
+                elif payload == OFF:
+                    self._state = False
+                # how do we handle the "overpower" state?
+            elif msg.topic == self._topic + "/power":
+
+                self._power = float(msg.payload)
+            elif msg.topic == self._topic + "/energy":
+                # convert from Watt-minute to kWh
+                self._energy = float(msg.payload) / 60000
+
+            self.async_write_ha_state()
+
+        self._unsubscribe = await self._mqtt.async_subscribe(
+            self._topic + "/#", state_message_received
+        )
+
+    async def async_will_remove_from_hass(self):
+        """Unsubscribe when removed."""
+        self._unsubscribe()
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        info = {
+            "identifiers": {(DOMAIN, self._unique_id)},
+            "manufacturer": "Shelly",
+            "model": self._model,
```

Comment:
```suggestion
            "model": MODELS[self._model][MODEL_TITLE],
```
This change will give us more user-friendly model names. We also need to import `MODEL_TITLE` from `const`.